src/recommendations/src/recommendations-service/app.py

```diff
@@ -433,16 +562,118 @@ def experiment_outcome():
         user_id = correlation_bits[1]
         variation_index = int(correlation_bits[2])
         result_rank = int(correlation_bits[3])
-        experiment.track_conversion(user_id = user_id, variation_index = variation_index, result_rank = result_rank)
+        experiment.track_conversion(user_id=user_id, variation_index=variation_index, result_rank=result_rank)
 
         return jsonify(success=True)
 
     except Exception as e:
         app.logger.exception('Unexpected error logging outcome', e)
-        raise BadRequest(message = 'Unhandled error', status_code = 500)
+        raise BadRequest(message='Unhandled error', status_code=500)
+
+
+def get_dataset_group_arn_from_name(dsg_name):
+    """Looks through the available dataset groups for one with the provided name and returns
+    its Arn."""
+    app.logger.info(f'Getting dataset group Arn')
+    dataset_group_arn = None
+    for dg in personalize.list_dataset_groups()['datasetGroups']:
+        if dg['name'] == dsg_name:
+            dataset_group_arn = dg['datasetGroupArn']
+    return dataset_group_arn
+
+
+def get_event_tracker_arn_from_id(dataset_group_arn, event_tracking_id):
+    """Looks for the Arn of an event tracker given its dataset group and Id"""
+    list_response = personalize.list_event_trackers(datasetGroupArn=dataset_group_arn, maxResults=100)
+    for event_tracker in list_response['eventTrackers']:
+        arn = event_tracker['eventTrackerArn']
+        describe_response = personalize.describe_event_tracker(eventTrackerArn=arn)
+        tracking_id = describe_response['eventTracker']['trackingId']
+        if tracking_id == event_tracking_id:
+            return arn
+    return None
+
+
+@app.route('/reset/realtime', methods=['POST'])
+def reset_realtime():
```

Comment:
This seems out of scope for the IVS integration. What is the requirement here?

Reply:
Since this demo is going to be running for a number of days at a conference provided functionality to reset the campaigns to their default state in case the output of the campaigns was to drift over the course of a number of demos. 

This reset functionality used to be possible with Amazon Personalize by deleting the so-called "EVENT_INTERACTIONS" dataset but this dataset seems deprecated so we experimented with reset options and ultimately needed to:
- Delete dataset group & all content
- Recreate dataset group with a new name
- Recreate solutions and campaigns with new names

Hopefully this provides some background for some of the other changes (e.g. why we merged delete and create functionality into one lambda).

Following yesterday's (22nd Oct) call we are also aware of the blue-green deployment approach that other teams have been working on - we should discuss this further on our next call so we can unify our approach to this sort of functionality.